packages/rfw/example/wasm/lib/platforms/desktop.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:wasm/wasm.dart';
+
+import 'platform.dart';
+
+// This file implements the Network and Wasm APIs through the wasm package
+// for desktop.
+
+class NetworkImplementation extends Network {
+  @override
+  Future<Uint8List> get(String url) async {
+    final HttpClientResponse client =
+        await (await HttpClient().getUrl(Uri.parse(url))).close();
+    return Uint8List.fromList(
+        await client.expand((List<int> chunk) => chunk).toList());
+  }
+}
+
+class WasmImplementation extends Wasm {
+  late final WasmInstance _wasmInstance;
+
+  @override
+  Future<void> loadModule(String url) async {
+    final wasmBytes = await NetworkImplementation().get(url);
```

Comment:
should this take a NetworkImplementation, rather than creating its own?

Reply:
probably premature optimization. Since things in this file are meant to be coupled for the platform.

Reply:
well mostly it seems weird to me to have an API that appears mockable but is not. If we expect loadModule to use the network, IMHO either the code that owns loadModule should also be the code that creates the NetworkImplementation, or the code should be given one.

Reply:
sure, made it inject a network into wasm